server/embedly.js

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+// Facade that abstracts the embed.ly endpoint.
+// TODO use Joi to validate the url and ensure key is defined
+
+var camelize = require('underscore.string').camelize;
+var Embedly = require('embedly');
+
+var config = require('./config');
+var apiKey = config.get('embedly.apiKey');
+var isEnabled = config.get('embedly.enabled');
+var logFactory = require('./logger');
+var log = logFactory('server.services.embedly');
+var npmLog = logFactory('server.services.embedly.vendor');
+
+module.exports = {
+  extract: function extract(url, cb) {
+    log.debug('embedly.extract called');
+
+    if (!isEnabled) {
+      return cb('Embedly is currently disabled. Set `server.embedly.enabled` to enable.');
+    }
+
+    // node-embedly logs failures, so we can skip that here
+    new Embedly({key: apiKey, logger: npmLog}, function(err, api) {
+      if (err) { return cb(err); }
+      api.extract({url:url}, function(err, data) {
+        log.verbose('embedly.extract callback for url ' + url + ' returned data ' + JSON.stringify(data));
+        if (err) { return cb(err); }
+        var d = data && data[0];
+        if (!d) { return cb(new Error('embedly response was empty for url ' + url)); }
+
+        // for right now, let's just return the keys we want in user_pages
+        var out = {};
+
+        // these keys are top-level and match the format in the database
```

Comment:
it'd be nice to split all these stanzas into tiny helper functions, would make them easier to test / isolate bugs